src/test/java/seedu/address/logic/commands/FindCommandTest.java

```diff
@@ -68,24 +79,116 @@ public void execute_zeroKeywords_noPersonFound() {
         NameContainsKeywordsPredicate namePredicate = prepareNamePredicate(" ");
         PhoneContainsKeywordsPredicate phonePredicate = preparePhonePredicate(" ");
         EmailContainsKeywordsPredicate emailPredicate = prepareEmailPredicate(" ");
+        RentalInformationContainsKeywordsPredicate rentalInfoPredicate = prepareRentalInfoPredicate(" ");
 
-        FindCommand command = new FindCommand(namePredicate, phonePredicate, emailPredicate);
-        expectedModel.updateFilteredPersonList(namePredicate.or(phonePredicate).or(emailPredicate));
+        FindCommand command = new FindCommand(namePredicate, phonePredicate, emailPredicate, rentalInfoPredicate);
+        expectedModel.updateFilteredPersonList(
+                namePredicate.or(phonePredicate).or(emailPredicate).or(rentalInfoPredicate));
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(Collections.emptyList(), model.getFilteredPersonList());
     }
 
-    //TODO: WRITE TEST CASE FOR THE BELOW MULTIPLE KEYWORDS IN THE FUTURE
+    @Test
+    public void execute_addressKeyword_success() {
+        String address = "BLK";
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+
+        NameContainsKeywordsPredicate namePredicate = prepareNamePredicate(address);
+        PhoneContainsKeywordsPredicate phonePredicate = preparePhonePredicate(address);
+        EmailContainsKeywordsPredicate emailPredicate = prepareEmailPredicate(address);
+        RentalInformationContainsKeywordsPredicate rentalInfoPredicate = prepareRentalInfoPredicate(address);
+
+        FindCommand command = new FindCommand(namePredicate, phonePredicate, emailPredicate, rentalInfoPredicate);
+        expectedModelWithRental.updateFilteredPersonList(
+                namePredicate.or(phonePredicate).or(emailPredicate).or(rentalInfoPredicate));
+        assertCommandSuccess(command, modelWithRental, expectedMessage, expectedModelWithRental);
+        assertEquals(expectedModelWithRental.getFilteredPersonList(), modelWithRental.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_monthlyRentKeyword_success() {
+        String address = "2900";
```

Comment:
monthlyRent can be a better variable name here.

Reply:
Thanks for pointing that out!